### PR TITLE
Blogs UI polish

### LINKED ISSUE #27: UI Polish Review: Blog template/section
### Final polish review
- [ ] Settings have been deblockified and should match what is done [here for the blog template](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=8749%3A11), and [here for the blog posts section](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=8697%3A79656)

**Blog cards**
- [ ] On desktop the current padding on the top and bottom is 4rem and left and right is 5rem. Can we change both to 3rem? [Screenshot](https://screenshot.click/21-46-c8vkh-3q5al.png)
- [ ] On mobile the padding is 2.5rem and 3rem. Can we all the padding around the card to 3rem? 
- [ ] Remove 'read more' link from blog cards. The whole card is clickable. [Screenshot](https://screenshot.click/21-07-tiw1c-fdag8.png)
- [ ] Can we add a non-breaking space between the number of comments and the "comments" label?
- [ ]  the blog post empty state is off: it should reflect the layout a bit more?[screenshot](https://user-images.githubusercontent.com/2348980/122498015-62018480-cfbc-11eb-93fd-0f78cea5dde5.png)

**Blog section**
- [ ] 'View all' should be aligned with the title of the section on mobile. [Screenshot](https://screenshot.click/21-49-4nx0f-40e44.png)
- [ ] Remove slider when there is only one blog. [Screenshot](https://screenshot.click/21-40-f4zc0-pldnj.png)

**Blog post page**
- [ ] Can we remove the underline from the back to blog button and have the hover state move the arrow with the same animation used for collection cards? [Screenshot of current design](https://screenshot.click/21-13-3pu2k-2p9bm.png)
- [ ] Remove [Social sharing options](https://screenshot.click/21-06-6ww6g-3gskq.png) from Theme settings and leverage the native web API on the Blog post template, just like on the product template.

### PR DESCRIPTION
**Pull request by kmeleta**
**Why are these changes introduced?**

Fixes #27 

**Blog cards**
- Change desktop padding from `4rem 5rem` to `3rem`
- Change mobile padding from `2.5rem 3rem` to `3rem`
- Remove 'read more' link from blog cards
- Add non-breaking space between the number and "comments" label? (This was fixed in another PR)
- Align empty post styles with existing section styles [screenshot](https://user-images.githubusercontent.com/2348980/122498015-62018480-cfbc-11eb-93fd-0f78cea5dde5.png)

**Main blog section**
- Deblockify section, move show date and show author to [section settings](https://screenshot.click/08-53-o27y3-l12dt.png)

**Featured blog section**
- Deblockify section, move show date and show author to [section settings](https://screenshot.click/08-54-0brjq-hyjra.png)
- 'View all' should be aligned with the title of the section on mobile (This was fixed in slider polish PR)
- Remove slider when there is only one blog (This was fixed in slider polish PR)

**Blog post/article page**
- Remove underline from the "back to blog" button and use same arrow animation as collection cards
- Replace social sharing settings with native web API, like on the product template
- (Related to above) Remove the now unused `social-sharing` snippet, theme settings, and translations [screenshot](https://screenshot.click/21-06-6ww6g-3gskq.png)

**Demo links**

- [Editor](https://os2-demo.myshopify.com/admin/themes/120850513942/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by melissaperreault — COMMENTED** [review]
Thank you Ken! 🎉 
I did a first pass and here are a few observations:

**Share block**
- [x] [The link on the fall back is not updated](https://screenshot.click/21-07-jo658-251ly.png) with the right page link to have direct access to the blog post.

**Blog post cards**
- [x] Spacing hierarchy needs additional love since the new typography changes, let's add a [1.2rem top margin](https://screenshot.click/21-07-lbbxx-1qfxg.png) on the excerpt text. This will make is safe for [when we have Date and Author on and off](https://screenshot.click/21-07-n4liq-hnrrv.png).
- [x] For this request "make it look more like an actual blog card" the guidance could have been more visual, sorry for the gap. [This is a good reference](https://screenshot.click/21-07-l4i2e-ueh2r.png), we just need to make it look like a blog post card and use the empty illustration placeholder just like on [Image with text ](https://screenshot.click/21-07-etzj0-rszci.png)
- [x] [Placeholder text color opacity](https://screenshot.click/21-07-lwkbt-6ncz5.png), we can bump it to 75% to match other sections.

**Date/Author**
- [ ] The line height [for this specific piece](https://screenshot.click/21-07-x2t3d-3gz26.png) is really too tall but I couldn't check where to adjust and tell a value, we can pair to find the right balance. This is for all blog instances.

Thank you!!

**Review by tauthomas01 — APPROVED** [review]
Code looks good. Let me know if you need another review if there are design changes.

File: sections/main-article.liquid [hunk 1]
```diff
@@ -48,16 +48,48 @@
...
+                <summary class="share-button__button button button--tertiary">
+                  {% render 'icon-share' %}
+                  {{ 'general.share.share' | t }}
+                </summary>
+                <div class="share-button__fallback">
+                  <div class="field">
+                    <input type="text"
+                          class="field__input"
+                          id="url"
+                          value="{{ shop.url | append: product.url }}"
```
Comment (ludoboludo) [inline sections/main-article.liquid:68]: Here we need to change the url to the article :) 
```suggestion
                          value="{{ shop.url | append: article.url }}"
```

**Review by ludoboludo — COMMENTED** [review]
Looking good. Left just a few comments.

File: sections/main-article.liquid (hunk 1 above)
Comment (kmeleta) [reply sections/main-article.liquid]: doh.

File: assets/section-featured-blog.css [hunk 2]
```diff
@@ -7,16 +7,17 @@
...
 .blog .placeholder {
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  height: 22rem;
-  text-align: center;
-  padding: 4rem 2rem 5rem;
-  margin: 0 2rem;
+  padding: 3rem;
+  margin: 0 1.5rem;
```
[the remark this answers is not in the document]
Comment (kmeleta) [reply assets/section-featured-blog.css:10]: Per Meli's guidance, added 1.2rem below the heading, but we decided to keep the paddings for the placeholder the same as the current article card. 3rem all around and the additional bottom margin from the `<p>`. Looks less weird now with the placeholder image and all.

**Review by melissaperreault — DISMISSED** [review]
Thank you Ken! 

- Just one minor thing to adjust for the [blog post template](https://screenshot.click/21-07-6vf1l-q58bs.png), to adjust the line height for the date and author as well and make it smaller.

File: assets/base.css [hunk 3]
```diff
@@ -1173,6 +1173,49 @@ deferred-media {
...
+
+.share-button__fallback {
+  border: 0.1rem solid rgba(var(--color-foreground), 0.1);
+  background: rgb(var(--color-background));
+  padding: 2rem;
+  position: absolute;
+  top: 4rem;
+  left: -0.1rem;
+  z-index: 3;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (kmeleta) [reply assets/base.css:1195]: Added 60% width (tablet and up) but only for the share block on the blog post page currently. (I assume the product page share block should continue to be 100%.) If there's somewhere this needs to be accounted for, let me know.

**Review by melissaperreault — DISMISSED** [review]
🚢 🚢 🚢 ! Looking good thanks!

File: assets/section-featured-blog.css [hunk 4]
```diff
@@ -7,21 +7,50 @@
...
+  overflow: hidden;
+  position: relative;
+  padding-bottom: 66.6%;
+}
+
+.blog .placeholder .placeholder-svg {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
```
Comment (ludoboludo) [inline assets/section-featured-blog.css:40]: I don't think you need this if you're doing `width: 100%; height: 100%;`.

File: sections/featured-blog.liquid [hunk 5]
```diff
@@ -62,12 +62,17 @@
...
     {%- else -%}
       <div class="placeholder">
-        <h2>
-          {{ 'sections.featured_blog.onboarding_title' | t }}
-        </h2>
-        <p class="rte-width">
-          {{ 'sections.featured_blog.onboarding_content' | t }}
-        </p>
+        <div class="placeholder-image">
+          {{ 'image' | placeholder_svg_tag: 'placeholder-svg' }}
```
Comment (ludoboludo) [inline sections/featured-blog.liquid:66]: I think you could add the `placeholder` class in here actually. I feel like that's more what it was meant for 🤔  And then the parent can use a new class maybe like `blog-placeholder`. That would also reduce some of the selectors in the CSS (`.blog .placeholder .placeholder-content` vs `blog-placeholder__content`) and keep it specific to this section which it is right now. 

What do you think ?

File: assets/section-featured-blog.css [hunk 6]
```diff
@@ -7,21 +7,50 @@
...
+.blog .placeholder .placeholder-content {
+  padding: 3rem;
+}
+
+.blog .placeholder .placeholder-image {
+  background: rgba(var(--color-foreground), .04);
+  overflow: hidden;
+  position: relative;
+  padding-bottom: 66.6%;
+}
```
Comment (ludoboludo) [inline assets/section-featured-blog.css:34]: This seems quite similar to the `.media` class. Could use it instead maybe: `media media--transparent media--landscape`

File: assets/section-featured-blog.css [hunk 7]
```diff
@@ -7,21 +7,50 @@
...
+
+.blog .placeholder .placeholder-svg {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 100%;
+  fill: currentColor;
+}
```
Comment (tauthomas01) [inline assets/section-featured-blog.css:43]: This look very similar to `media` component class. Could we leverage that along with `.media--landscape`?

File: assets/section-featured-blog.css [hunk 8]
```diff
@@ -7,21 +7,50 @@
...
+
+.blog .placeholder h2 {
+  margin: 0;
+}
+
+.blog .placeholder .rte-width {
```
Comment (tauthomas01) [inline assets/section-featured-blog.css:49]: Is it possible to reduce specificity if we use `.blog .rte-width`?

**Review by tauthomas01 — APPROVED** [review]
I have some small comments regarding the CSS code, but the rest looks good.

File: sections/featured-blog.liquid (hunk 5 above)
Comment (kmeleta) [reply sections/featured-blog.liquid:66]: hmm, yeah I see what you mean. To me `.placeholder` doesn't really seem intuitive as representing just the image in the case of placeholder image _and_ text like we have here, but in context, what you suggest is the best bet. Updated.

File: assets/section-featured-blog.css (hunk 7 above)
Comment (kmeleta) [reply assets/section-featured-blog.css]: Using .media and .media--landscape, does save a couple of lines here and makes sense to utilize. I currently still need to include the centering styles though (see https://github.com/Shopify/dawn/pull/152#discussion_r669793547)

**Review by ludoboludo — APPROVED** [review]
LGTM 👌

**Pull request merged**